test/unit/oidc_jwt_token_validate_test.rb

```diff
@@ -0,0 +1,72 @@
+require 'test_helper'
+require 'ostruct'
+require 'openssl'
+require 'jwt'
+
+class OidcJwtTokenValidateTest < ActiveSupport::TestCase
+  context '#valid_jwt?' do
+    def setup
+      skip "SSO feature is not available for Ruby < 2.4.0" unless
+        OpenSSL::PKey::RSA.new.respond_to?(:set_key)
+      @jwk = JWT::JWK.new(OpenSSL::PKey::RSA.new(2048))
```

Comment:
Generating private keys is slow, especially when entropy is low (common on VMs). This can easily blow up the test runtime. I'd consider a fixture.